pkg/registry/apis/dashboard/access/sql_dashboards.go

```diff
@@ -84,34 +85,78 @@ const selector = `SELECT
   LEFT OUTER JOIN user AS UpdatedUSER ON dashboard.created_by = UpdatedUSER.id
   WHERE is_folder = false`
 
-// GetDashboards implements DashboardAccess.
-func (a *dashboardSqlAccess) GetDashboards(ctx context.Context, query *DashboardQuery) (*dashboardsV0.DashboardList, error) {
+func (a *dashboardSqlAccess) getRows(ctx context.Context, query *DashboardQuery, onlySummary bool) (*rowsWrapper, int, error) {
+	if !query.Labels.Empty() {
+		return nil, 0, fmt.Errorf("label selection not yet supported")
+	}
+	if len(query.Requirements.SortBy) > 0 {
+		return nil, 0, fmt.Errorf("sorting not yet supported")
+	}
+	if query.Requirements.ListHistory != "" {
+		return nil, 0, fmt.Errorf("ListHistory not yet supported")
+	}
+	if query.Requirements.ListDeleted {
+		return nil, 0, fmt.Errorf("ListDeleted not yet supported")
+	}
+	if len(query.Requirements.ListOriginKeys) > 0 {
+		return nil, 0, fmt.Errorf("ListOriginKeys not yet supported")
+	}
+
 	token, err := readContinueToken(query)
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 
 	limit := query.Limit
 	if limit < 1 {
 		limit = 15 //
 	}
+	args := []any{query.OrgID}
 
-	rows, err := a.doQuery(ctx, selector+`
-		AND dashboard.org_id=$1
-		AND dashboard.id>=$2
-		ORDER BY dashboard.id asc
-		LIMIT $3
-	`, query.OrgID, token.id, (limit + 2))
+	sqlcmd := selector
+
+	// We can not do this yet because title + tags are in the body
+	if onlySummary && false {
+		sqlcmd = strings.Replace(sqlcmd, "dashboard.data", `"{}"`, 1)
+	}
+
+	sqlcmd = fmt.Sprintf("%s AND dashboard.org_id=$%d", sqlcmd, len(args))
```

Comment:
I'm not wild about this style query builder... but OK for now

Reply:
Yeah, I think as long as we use a prepared statement in the end we shouldn't worry too much.